course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+#include <cstring>
+#include "my_string.h"
+
+MyString::MyString(){
+	buffer = std::unique_ptr<char[]>(new char[1]);
+	buffer = '\0';
```

Comment:
```buffer = std::unique_ptr(new char('\0'));```
will fail, because there are two types of allocation: single object allocation (new/delete), and array allocation (new[N]/delete[]).
This form of unique pointer you use here (```std::unique_ptr<char[]>```) works with array allocation, while ```new char``` allocates single  char, so it's a single object allocation.